File: C#/Calculations/PaintCalculator.cs
using static CSharp.DataValidation;
namespace CSharp.Calculations
{
    internal class PaintCalculator
    {
        private class PaintGallons
        {
            private const short ONE_GALLON = 350; // Square Feet
            private readonly float area;
            private readonly float gallons;

            public PaintGallons(string _roomType)
            {
                if (_roomType == "rectangular")
                {
                    float length = ValidatedNumber<float>("Length: ");
                    float width = ValidatedNumber<float>("Width: ");
                    area = length * width;
                }
                else if (_roomType == "round")
                {
                    float radius = ValidatedNumber<float>("Radius: ");
                    area = (float)(Math.PI * radius);
                }
                else if (_roomType == "l-shaped")
                {
                    float length1 = ValidatedNumber<float>("Length of the first rectangle: ");
                    float width1 = ValidatedNumber<float>("Width of the first rectangle: ");
                    float area1 = length1 * width1;

                    float length2 = ValidatedNumber<float>("Length of the second rectangle: ");
                    float width2 = ValidatedNumber<float>("Width of the second rectangle: ");
                    float area2 = length2 * width2;

                    area = area1 + area2;
                }

                gallons = (float)Math.Ceiling(area / ONE_GALLON);
                string plurality = gallons <= 1 ? "gallon" : "gallons";
                Display(plurality);
            }

            private void Display(string plurality)
            {
                Console.WriteLine($"You will need to purchase {gallons} {plurality} of paint to cover {area:F0} square feet.");
            }
        }

        public static void Main7()
        {
            Console.WriteLine("[1] Square/Rectangular");
            Console.WriteLine("[2] Round");
            Console.WriteLine("[3] L-Shaped");
            sbyte choice = ValidatedNumber<sbyte>("Enter your room type: ", _min: 1, _max: 3);
            if (choice == 1)
                _ = new PaintGallons("rectangular");
            else if (choice == 2)
                _ = new PaintGallons("round");
            else if (choice == 3)
                _ = new PaintGallons("l-shaped");
        }
    }
}

File: Python/Calculations/PaintCalculator.py
# Revise the program to ensure that inputs are entered as numeric values.
# Don't allow the user to proceed if the value entered is not numeric.
from DataValidation import ValidatedNumber
import math

# Calculate gallons of paint needed to paint the ceiling of a room.
# Prompt for the length and width, and assume one gallon covers 350 square feet.
# Display the number of gallons needed to paint the ceiling as a whole number.

# Use a constant to hold the conversion rate.
ONE_GALLON = 350  # square feet


class PaintGallons:
    def __init__(self, room_type=None):
        if room_type == "square":
            self.length = ValidatedNumber("Length: ")
            self.width = ValidatedNumber("Width: ")
            self.area = self.length * self.width
        # Implement support for a round room.
        elif room_type == "round":
            self.radius = ValidatedNumber("Radius: ")
            self.area = math.pi * (self.radius ** 2)
        # Implement support for an L-shaped room.
        elif room_type == "l-shaped":
            self.length1 = ValidatedNumber("Length of the first rectangle: ")
            self.width1 = ValidatedNumber("Width of the first rectangle: ")
            self.area1 = self.length1 * self.width1

            self.length2 = ValidatedNumber("Length of the second rectangle: ")
            self.width2 = ValidatedNumber("Width of the second rectangle: ")
            self.area2 = self.length2 * self.width2

            self.area = self.area1 + self.area2

        # Ensure that you round up to the next whole number.
        self.gallons = math.ceil(self.area / ONE_GALLON)
        plurality = self.gallons <= 1 and "gallon" or "gallons"
        print(f"You will need to purchase {self.gallons} {plurality} of paint to cover {self.area} square feet.")


print("[1] Square/Rectangular")
print("[2] Round")
print("[3] L-Shaped")
choice = ValidatedNumber("Enter your room type: ", 1, 3)
if choice == 1:
    paint_buckets = PaintGallons("square")
elif choice == 2:
    paint_buckets = PaintGallons("round")
elif choice == 3:
    paint_buckets = PaintGallons("l-shaped")
